src/atom_finder/classifier/implicit-predicate.clj

```diff
@@ -0,0 +1,38 @@
+(in-ns 'atom-finder.classifier)
+(import '(org.eclipse.cdt.core.dom.ast  IASTIfStatement IASTForStatement IASTWhileStatement IASTDoStatement IASTBinaryExpression IASTUnaryExpression))
+
+(def implicit-types #{IASTIfStatement IASTForStatement IASTWhileStatement IASTDoStatement IASTConditionalExpression})
+; If statement as an implicit type:  C99 6.8.4.1/1
+; For/While/Do statements as implicit types:  C99 6.8.5/2
+; Ternary operator as an implicit type:  C99 6.5.15/1
+
+(declare implicit-expression?)
+
+(defn remove-parentheses
+[expr]
+(cond
+  (== IASTUnaryExpression/op_not (.getOperator expr)) true
+  (== IASTUnaryExpression/op_bracketedPrimary (.getOperator expr))
+    (not (implicit-expression? (.getOperand expr)))
+  :else false))

```

Comment:
It doesn't look like this function removes parentheses, it looks like applies heuristics to unary operators. If instead you made it actually remove parentheses then you wouldn't need the cyclic reference either, you could just do something like:

(->> node remove-parentheses implicit-expression?)

Also, fix your indentation.